### ISSUE TITLE
Program does not terminate after closing connection to replica set

### ISSUE THREAD

**Issue by TheNilesh** [sboesebeck/morphium#14]
The morphium starts non-daemon thread. Which prevents normal program termination.

    public static void main(String args[]) throws UnknownHostException {
		MorphiumConfig cfg = new MorphiumConfig();
		cfg.setDatabase("mydb");
		cfg.setHosts("10.10.10.17:27015,10.10.10.15:27015");
                morphium = new Morphium(cfg);
		
		System.out.println("Closing Connection");
		morphium.close();
		System.out.println("Main Exits");
		//System.exit(0);
	}

*Output:*
```
Closing Connection
Jan 11, 2016 6:11:18 PM com.mongodb.diagnostics.logging.JULLogger log
INFO: Opened connection [connectionId{localValue:3}] to 10.10.10.15:27015
Jan 11, 2016 6:11:18 PM com.mongodb.diagnostics.logging.JULLogger log
INFO: Closed connection [connectionId{localValue:3}] to 10.10.10.15:27015 because the pool has been closed.
Jan 11, 2016 6:11:18 PM com.mongodb.diagnostics.logging.JULLogger log
INFO: Closed connection [connectionId{localValue:1}] to 10.10.10.17:27015 because the pool has been closed.
Main Exits
```
JVM is still running, I need to press Ctrl+C to exit.
------------------------------------------------
This happens only in case of replica set.
morphium : 2.2.23
mongo-java-driver : 3.0.3

**Comment by sboesebeck** [comment]
Hi,

which Version of Morphium are you using - this should be fixed already.

Do you call close() on morphium?

cheers,

Stephan

**Comment by TheNilesh** [comment]
_morphium-2.2.23.jar_ via maven.

**Comment by sboesebeck** [comment]
Thanks for bringing that up - there is a problem with some executors... fixed that for 2.2.24 - will create a bugfixrelease soon...

I'll post a Pre-Relase today, which you can use directly from github, if you want...

Cheers,

Stephan

Am 12.01.2016 um 06:47 schrieb niLesh Akhade <notifications@github.com <mailto:notifications@github.com>>:

**Comment by sboesebeck** [comment]
Fixed in Pre-Release V2.2.24BETA1 - when there are no errors, things will be pushed to maven...
